tests/functional/api/BackbeatServer.js

```diff
@@ -460,11 +485,31 @@ describe('Backbeat Server', () => {
                     setHash(redisClient, keys, next);
                 }, err => {
                     assert.ifError(err);
-                    getRequest('/_/crr/failed', (err, res) => {
-                        assert.ifError(err);
-                        assert.strictEqual(res.length, 2000 * 5);
-                        done();
-                    });
+                    const keyCount = 2000 * 5;
+                    const scanCount = 1000;
+                    let marker = 0;
+                    const sites = [];
+                    async.timesSeries(keyCount / scanCount, (i, next) =>
+                        getRequest(`/_/crr/failed?marker=${marker}`,
+                        (err, res) => {
+                            assert.ifError(err);
+                            res.Versions.forEach(version => {
+                                // Ensure we have no duplicate results.
+                                assert(!sites.includes(version.StorageClass));
```

Comment:
Not a strong concern, but the check has linear complexity, so the whole process is O(n^2), with 10000 elements it may slow down the test quite a bit. An easy fix is to use a `Set` object and use `add()`, `has()` and `size` on it.

Reply:
Thanks for that tip, I was actually not aware of `Set`. I've updated as such.